rotkehlchen/history/events/structures/asset_movement.py

```diff
@@ -0,0 +1,210 @@
+import logging
+from enum import Enum, auto
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.accounting.mixins.event import AccountingEventType
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.location_details import get_formatted_location_name
+from rotkehlchen.exchanges.data_structures import hash_id
+from rotkehlchen.history.events.structures.base import (
+    HISTORY_EVENT_DB_TUPLE_WRITE,
+    HistoryBaseEntry,
+    HistoryBaseEntryType,
+)
+from rotkehlchen.history.events.structures.types import (
+    HistoryEventSubType,
+    HistoryEventType,
+)
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_fval
+from rotkehlchen.types import (
+    Location,
+    TimestampMS,
+)
+from rotkehlchen.utils.misc import ts_ms_to_sec
+
+if TYPE_CHECKING:
+    from more_itertools import peekable
+
+    from rotkehlchen.accounting.mixins.event import AccountingEventMixin
+    from rotkehlchen.accounting.pot import AccountingPot
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class AssetMovementCategory(Enum):
```

Comment:
Why do you re-declare this? This already exists at rotkehlchen/types.py.  Though you don't really need his here at all

Reply:
Removed this, so its kind of irrelevant now, but just to explain my reason - I thought it would be better to use a plain enum rather than using the existing DBCharEnumMixIn... but I suppose that would have been converted to a plain enum later when the db part gets removed anyway - so yeah, I guess this was unnecessary either way.